modules/core/src/main/java/org/apache/synapse/config/xml/EnrichMediatorFactory.java

```diff
@@ -187,4 +194,35 @@ public QName getTagQName() {
         return XML_Q;
     }
 
+    private void validateTypeCombination(OMElement sourceEle, OMElement targetEle) {
+        int sourceType = -1;
+        int targetType = -1;
+
+        // source type attribute
+        OMAttribute sourceTypeAttr = sourceEle.getAttribute(ATT_TYPE);
+        if (sourceTypeAttr != null && sourceTypeAttr.getAttributeValue() != null) {
+            sourceType = convertTypeToInit(sourceTypeAttr.getAttributeValue());
+            if (sourceType < 0) {
+                logger.info("Un-expected source type");
+            }
+        }
+
+        // target type attribute
+        OMAttribute targetTypeAttr = targetEle.getAttribute(ATT_TYPE);
+        if (targetTypeAttr != null && targetTypeAttr.getAttributeValue() != null) {
+            targetType = convertTypeToInit(targetTypeAttr.getAttributeValue());
+            if (targetType < 0) {
```

Comment:
It was the existing implementation, so need to change in many classes. Clear comments are added to explaining what this condition really means.